### ISSUE TITLE
[Improvement]: Add SSL support in Redis connector

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-extended-library#558]
### Connector Name

module/redis (Redis)

### Suggested improvement

Currently, the Redis connector does have any client configurations for SSL /TLS connections.

To add the support we have to  follow the following implementation

https://github.com/lettuce-io/lettuce-core/wiki/SSL-Connections

**Comment by malinthar** [comment]
I used the connector version 2.4.0 to run a Ballerina sample with the following client configuration.

```Ballerina
 redisConfig = {
            host:  "sample-dns.redis.cache.windows.net:6380",
            password: redisSingleHostPassword,
            options: {
                connectionPooling: false,
                isClusterConnection: false,
                ssl: true,
                startTls: false,
                verifyPeer: true,
                connectionTimeout: 1000000
            }
        };
```

Got the following stack-trace when trying to make a SSL Connection to a redis-instance.
```
Caused by: java.lang.NoClassDefFoundError: io/netty/channel/unix/UnixChannel        at io.netty.handler.ssl.SslHandler.setOpensslEngineSocketFd(SslHandler.java:2142)        at io.netty.handler.ssl.SslHandler.handlerAdded(SslHandler.java:1981)

```

This behaviour is due to an incompatible netty version. The connector  2.4.0 uses lettuce-core 5.1.2 dependency which is compatible only with `netty-1.4.30`[1]. However, the connector includes netty-1.4.86 dependency [2]  and this causes the ClassNotFoundException in lettuce-core due to incompatible netty versions.

When the netty version in the connector [2] was downgraded  to 1.4.20-Final the program with the client configuration above works as expected.

In summary,  the netty jar version included in the connector pack is not compatible with lettuce-core. The minimal compatible version of lettuce-core with netty-1.4.86 is 6.2.3 release[3]. If we are to bump the lettuce-core verion  this would introduce a lot of API changes and would take a considerable amount of time to  migrate. 

@NipunaRanasinghe how should we proceed. Downgrading the netty-version in the connector or upgrading the lettuce-core verion?

[1] https://github.com/lettuce-io/lettuce-core/blob/6567767c41fcea5002a1a2ec200ea71f867f6d8d/pom.xml#L61
[2] https://github.com/ballerina-platform/module-ballerinax-redis/blob/a6a82781ee08fde15e050d8359b158647b123770/redis/pom.xml#L59
[3] https://github.com/lettuce-io/lettuce-core/blob/318c6aa4f6146ff047f824eb25ff53cc3b68b299/pom.xml#L64

**Comment by NipunaRanasinghe** [comment]
@malinthar considering the urgency of the fix and potential compatibility issues, downgrading the netty-version would be the safer option IMO. 
Of course we should also work on migrate to the latest lettuce version, to support all the latest improvements and fixes from the lettuce core, which I think we can align with our next minor version release as a separate approach

**Comment by malinthar** [comment]
> @malinthar considering the urgency of the fix and potential compatibility issues, downgrading the netty-version would be the safer option IMO. Of course we should also work on migrate to the latest lettuce version, to support all the latest improvements and fixes from the lettuce core, which I think we can align with our next minor version release as a separate approach

@NipunaRanasinghe Ack! Let's proceed with downgrading

**Comment by malinthar** [comment]
We have released https://central.ballerina.io/ballerinax/redis/2.4.1 with netty 1.4.30. However, http services stoped working with the following error after this change. 

![image](https://github.com/ballerina-platform/ballerina-extended-library/assets/35211477/823a36cf-ad69-4497-8008-b3fe40111b40)

This is because downgrading netty version  to `1.4.30-Final ` will not work as it is conflicting with the netty version (1.4.94) used by http (2.8.x) . According to netty documentation we can include the `netty-transport-native-unix-common-4.1.86.Final` to resolve the original issue. 

Therefore, we have  upgraded the netty version back to `1.4.86-Final` with https://github.com/ballerina-platform/module-ballerinax-redis/pull/175

**Comment by malinthar** [comment]
> We have released https://central.ballerina.io/ballerinax/redis/2.4.1 with netty 1.4.30. However, http services stoped working with the following error after this change.
> 
> ![image](https://user-images.githubusercontent.com/35211477/263614977-823a36cf-ad69-4497-8008-b3fe40111b40.png)
> 
> This is because downgrading netty version to `1.4.30-Final ` will not work as it is conflicting with the netty version (1.4.94) used by http (2.8.x) . According to netty documentation we can include the `netty-transport-native-unix-common-4.1.86.Final` to resolve the original issue.
> 
> Therefore, we have upgraded the netty version back to `1.4.86-Final` with [ballerina-platform/module-ballerinax-redis#175](https://github.com/ballerina-platform/module-ballerinax-redis/pull/175)

released https://central.ballerina.io/ballerinax/redis/2.4.1 with the fix
